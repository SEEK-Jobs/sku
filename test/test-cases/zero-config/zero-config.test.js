const dirContentsToObject = require('../../utils/dirContentsToObject');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const waitForUrls = require('../../utils/waitForUrls');
const getAppSnapshot = require('../../utils/getAppSnapshot');

describe('zero-config', () => {
  beforeAll(async () => {
    await runSkuScriptInDir('configure', __dirname);
  });

  describe('start', () => {
    const devServerUrl = `http://localhost:8080`;
    let server;

    beforeAll(async () => {
      server = await runSkuScriptInDir('start', __dirname);
      await waitForUrls(devServerUrl);
    });

    afterAll(() => {
      server.kill();
    });

    it('should start a development server', async () => {
      const snapshot = await getAppSnapshot(devServerUrl);
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('build', () => {
    beforeAll(async () => {
      await runSkuScriptInDir('build', __dirname);
    });

    it('should generate the expected files', async () => {
      const files = await dirContentsToObject(`${__dirname}/dist`);
      expect(files).toMatchSnapshot();
    });
  });

  describe('format', () => {
    it('should format successfully', async () => {
      const { childProcess } = await runSkuScriptInDir('format', __dirname);
      expect(childProcess.exitCode).toEqual(0);
    });
  });

  describe('lint', () => {
    it('should lint successfully', async () => {
      const { childProcess } = await runSkuScriptInDir('lint', __dirname);
      expect(childProcess.exitCode).toEqual(0);
    });
  });
});
