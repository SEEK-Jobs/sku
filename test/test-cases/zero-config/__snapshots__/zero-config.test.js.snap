// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zero-config build should generate the expected files 1`] = `
Object {
  "0-2d4c17dcf56e1762b025.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "index.html": "<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>hello-world</title>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    
  </head>
  <body>
    <div id=\\"app\\" data-message=\\"Hello world!\\"></div>
    <script type=\\"text/javascript\\" src=\\"/runtime-f2546d261e10324b4552.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/main-2d0ef90d595da2b109cf.js\\"></script>
  </body>
</html>",
  "main-2d0ef90d595da2b109cf.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime-f2546d261e10324b4552.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;

exports[`zero-config start should start a development server 1`] = `
Object {
  "content": "<!DOCTYPE html><html><head>
    <meta charset=\\"UTF-8\\">
    <title>hello-world</title>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    
  <script charset=\\"utf-8\\" src=\\"/0.js\\"></script></head>
  <body>
    <div id=\\"app\\" data-message=\\"Hello world!\\">Hello world!</div>
    <script type=\\"text/javascript\\" src=\\"/runtime.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/vendors~main.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/main.js\\"></script>
  
</body></html>",
  "errors": Array [],
  "sourceHtml": "<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>hello-world</title>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    
  </head>
  <body>
    <div id=\\"app\\" data-message=\\"Hello world!\\"></div>
    <script type=\\"text/javascript\\" src=\\"/runtime.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/vendors~main.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/main.js\\"></script>
  </body>
</html>",
  "warnings": Array [],
}
`;
